packages/component-base/test/url-utils.test.js

```diff
@@ -0,0 +1,33 @@
+import { expect } from '@esm-bundle/chai';
+import { arePathsMatching } from '../src/url-utils.js';
+
+describe('url-utils', () => {
+  describe('arePathsMatching', () => {
+    const paths = ['', '/', '/path', 'base/path'];
+
+    it('should return true when paths match', () => {
+      paths.forEach((path) => expect(arePathsMatching(path, path)).to.be.true);
+    });
+
+    it('should return false when paths do not match', () => {
+      expect(arePathsMatching('/path1', '/path2')).to.be.false;
+    });
+
+    it('should ignore leading and trailing spaces in paths', () => {
+      paths.forEach((path) => {
+        const pathWithExtraSpace = ` ${path} `;
+        expect(arePathsMatching(pathWithExtraSpace, path)).to.be.true;
+        expect(arePathsMatching(path, pathWithExtraSpace)).to.be.true;
+      });
+    });
+
+    it('should ignore a leading slash in paths', () => {
+      paths.forEach((path) => {
+        const pathWithoutLeadingSlash = path.startsWith('/') ? path.substring(1) : path;
+        const pathWithLeadingSlash = path.startsWith('/') ? path : `/${path}`;
+        expect(arePathsMatching(pathWithoutLeadingSlash, pathWithLeadingSlash)).to.be.true;
+        expect(arePathsMatching(pathWithLeadingSlash, pathWithoutLeadingSlash)).to.be.true;
+      });
+    });
```

Comment:
Let's also add a test to verify that the helper uses `document.baseURI`. Here is an example of how it might look like:

```
const stub = sinon.stub(document, 'baseURI');
stub.returns('/base');

expect(arePathsMatching('/base/path', 'path')).to.be.true;
```